### ISSUE TITLE
Service disabled checking ignores systemd masked units

### ISSUE THREAD

**Issue by comps** [ComplianceAsCode/content#4386]
(May be related to #2632.)

TL;DR: `xccdf_org.ssgproject.content_rule_service_kdump_disabled` reports failure when the service is masked instead of being disabled. My expected result (based on what the OVAL already checks for) would be a reported success as the service cannot start.

Looking at `shared/templates/template_OVAL_service_disabled`, the checking seems to be based on wheter `multi-user.target` "wants" the given service, which is the same as `systemctl enable/disable <service>`.
```
$ systemctl disable kdump
Removed symlink /etc/systemd/system/multi-user.target.wants/kdump.service.

$ systemctl enable kdump
Created symlink from /etc/systemd/system/multi-user.target.wants/kdump.service to /usr/lib/systemd/system/kdump.service.

$ ls -l /etc/systemd/system/multi-user.target.wants/kdump.service
lrwxrwxrwx. 1 root root 37 May 31 03:19 /etc/systemd/system/multi-user.target.wants/kdump.service -> /usr/lib/systemd/system/kdump.service
```
Masking, however, is different; it doesn't touch the symlink above, it instead creates a dummy service file with higher priority (`/etc` overrides `/usr/lib`) and makes it point to `/dev/null` so that systemctl literally cannot do any operations on the unit file other than unmasking.
```
$ systemctl mask kdump
Created symlink from /etc/systemd/system/kdump.service to /dev/null.

$ systemctl cat kdump
# /dev/null

$ systemctl is-enabled kdump
masked

$ systemctl unmask kdump
Removed symlink /etc/systemd/system/kdump.service.

$ systemctl cat kdump
# /usr/lib/systemd/system/kdump.service
[Unit]
Description=Crash recovery kernel arming
<snip>
```
As you can see, a service can be both "enabled" (started by `multi-user.target`) and "masked" at the same time - the enable/disable is like an ON/OFF switch, masking is like removing the power supply. In either case, the service is not automatically started.

Additionally, the big difference is that **disabled services can still be started** by the sysadmin or as a dependency and some certifications might need the "never start" guarantee that masked gives.

...

Therefore I propose:
 1. Modifying `template_OVAL_service_disabled` so that it detects masked services (`/etc/systemd/system/<servicename>.service` is a symlink to `/dev/null`)
 1. Adding a profile variable for `template_BASH_service_disabled` that would allow to specify whether a service should be just disabled or masked as well

PS: Some modification to `systemd_ocil_service_disabled` may also be needed.

Thanks!

**Comment by stevegrubb** [comment]
For security, we need to have hard guarantees that a service that is not supposed to run cannot be started by dependency or the system admin on a whim. Otherwise we have unreliable security assumptions. The tests for disabled should simply be migrated to test for masked. The prose should reflect that masked is the correct state of the services being banned. Cheers.

**Comment by shawndwells** [comment]
Seems ideal to update the XCCDF/OVAL/remediations to ``systemctl mask`` instead of ``systemctl disable``.

**Comment by yuumasato** [comment]
I agree with moving to masking the service instead of disabling it.
